### PR TITLE
Closes #29 - Allow passing of custom params to step template

### LINKED ISSUE #29: Allow passing of custom params to step template
I should be able to define custom key/values in step config and these be made available in the template

step config

``` js
'/stepName': {
  meta: {
    key: 'value'
  }
}
```

template

``` html
<div class="{{key}}"></div>
```

### PR DESCRIPTION
**Pull request by joefitter**
- Added to locals method to check step config for meta object, if present expose add all keys to locals for use in template
- Added docs including usage in README
- This is to generate templates programmatically and avoid repeating code

### CODE REVIEW

File: lib/base-controller.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ var Controller = require('hmpo-form-wizard').Controller;
 var util = require('util');
 var _ = require('lodash');
 
-var BaseController = function BaseController() {
-  this.confirmStep = '/confirm';
+var BaseController = function BaseController(options) {
+  this.confirmStep = options.confirmStep || '/confirm';
```
Comment (easternbloc) [inline lib/base-controller.js:8]: 👍

**Comment by easternbloc** [comment]
> I worry a little bit about moving meta to locals without checking. Can we keep the meta namespace to avoid collisions?

**Comment by joefitter** [comment]
My thoughts around this were if you are providing custom params to the template, you would likely want them to override any clashing locals provided further up the tree, perhaps `meta` isn't the best name, @JoeChapman suggested `overrides`?

**Comment by easternbloc** [comment]
if that's the case how about calling it 'locals' :)

**Comment by joefitter** [comment]
Good idea :) We went from `modelData` to `meta` to `overrides` but `locals` makes much more sense

**Comment by JoeChapman** [comment]
LGTM @easternbloc if you're happy with this, You/I can merge.

**Comment by easternbloc** [comment]
LGTM

**Pull request merged**